src/streamlink/webbrowser/chromium.py

```diff
@@ -0,0 +1,178 @@
+            # Disables client-side phishing detection
+            "--disable-client-side-phishing-detection",
+            # Disable some built-in extensions that aren't affected by `--disable-extensions`
+            "--disable-component-extensions-with-background-pages",
+            # Don't update the browser 'components' listed at chrome://components/
+            "--disable-component-update",
+            # Disable installation of default apps
+            "--disable-default-apps",
+            # Disable all chrome extensions
+            "--disable-extensions",
+            # Hide toolbar button that opens dialog for controlling media sessions
+            "--disable-features=GlobalMediaControls",
+            # Disable the "Chrome Media Router" which creates some background network activity to discover castable targets
+            "--disable-features=MediaRouter",
+            # Disables Chrome translation, both the manual option and the popup prompt
+            "--disable-features=Translate",
+            # Suppresses hang monitor dialogs in renderer processes
+            #   This flag may allow slow unload handlers on a page to prevent the tab from closing
+            "--disable-hang-monitor",
+            # Disables logging
+            "--disable-logging",
+            # Disables the Web Notification and the Push APIs
+            "--disable-notifications",
+            # Disable popup blocking. `--block-new-web-contents` is the strict version of this
+            "--disable-popup-blocking",
+            # Reloading a page that came from a POST normally prompts the user
+            "--disable-prompt-on-repost",
+            # Disable syncing with Google
+            "--disable-sync",
+            # Forces the maximum disk space to be used by the disk cache, in bytes
+            "--disk-cache-size=0",
+            # Disable reporting to UMA, but allows for collection
+            "--metrics-recording-only",
+            # Mute any audio
+            "--mute-audio",
+            # Disable the default browser check, do not prompt to set it as such
+            "--no-default-browser-check",
+            # Disables all experiments set on about:flags
+            "--no-experiments",
+            # Skip first run wizards
+            "--no-first-run",
+            # Disables the service process from adding itself as an autorun process
+            #   This does not delete existing autorun registrations, it just prevents the service from registering a new one
+            "--no-service-autorun",
+            # Avoid potential instability of using Gnome Keyring or KDE wallet
+            "--password-store=basic",
+            # No initial CDP target (no empty default tab)
+            "--silent-launch",
+            # Use mock keychain on Mac to prevent the blocking permissions dialog asking:
+            #   Do you want the application "Chromium.app" to accept incoming network connections?
+            "--use-mock-keychain",
```

Comment:
The rest of the options look good here, this is the only one I'm not sure about as it's a bit of a security bypass. Probably won't matter to 99.9% of users though.

Reply:
No idea what you mean here. See the links above that list where the parameter is described in more detail. This prevents annoying unrelated popups on macOS, nothing more. This has to run as transparently as possible. There is no security issue, because a completely new and empty user dir gets used.